rust-lib/src/lib.rs

```diff
@@ -120,36 +125,64 @@ where
     }
 }
 
-#[derive(Debug, Deserialize)]
-struct StartRgbArgs {
-    #[serde(with = "serde_with::rust::display_fromstr")]
-    network: bp::Chain,
-    #[serde(with = "serde_with::rust::display_fromstr")]
-    stash_endpoint: SocketLocator,
-    contract_endpoints: HashMap<ContractName, String>,
-    threaded: bool,
-    datadir: String,
+#[derive(Debug, Display, From, Error)]
+#[display(doc_comments)]
+#[non_exhaustive]
+enum RequestError {
+    /// String error
+    #[from]
+    StringErr(String),
+
+    /// Input value is not a JSON object
```

Comment:
Here you can do stuff like
```suggestion
    /// Input value is not a JSON object or JSON parse error: {_0}
```
which will provide more error details from the embedded error type

Reply:
cool, thanks, done